dj/api/metrics.py

```diff
@@ -81,3 +84,46 @@ async def read_metrics_sql(
         database_id=optimal_database.id,
         sql=str(query_ast),
     )
+
+
+@router.get("/metrics/common/dimensions/", response_model=List[str])
+async def common_dimensions(
+    metric: List[str] = Query(
+        title="List of metrics to find common dimensions for",
+        default=[],
+    ),
+    session: Session = Depends(get_session),
+) -> List[str]:
+    """
+    Return common dimensions for a set of metrics.
+    """
+    metric_nodes = []
+    errors = []
+    for node_name in metric:
+        statement = select(Node).where(Node.name == node_name)
+        try:
+            node = session.exec(statement).one()
+            if node.type != NodeType.METRIC:
+                errors.append(
+                    DJError(
+                        message=f"Not a metric node: {node_name}",
+                        code=ErrorCode.NODE_TYPE_ERROR,
+                    ),
+                )
+            metric_nodes.append(node)
+        except NoResultFound:
+            errors.append(
+                DJError(
+                    message=f"Metric node not found: {node_name}",
+                    code=ErrorCode.UNKNOWN_NODE,
+                ),
+            )
+
+    if errors:
+        raise DJException(errors=errors)
+
+    common = set(get_dimensions(metric_nodes[0]))
```

Comment:
This function only gets the first layer of dimensions right? We can probably update this to pull in further layers once there's better dimensions metadata, so that we're not pulling in a lot of extraneous dims.

Reply:
Currently this pulls the same metadata as the metrics/ endpoint (which uses the same function). You've made me realize it will actually think similarly named columns on different metric parent nodes are a "common dimension" which I don't think is a safe assumption. For now I'm going to keep it this way--I'll start looking at how we can enable the sql and data endpoints to accept multiple metrics at once and build the appropriate SQL. That will inevitably add some finer logic for this.